sound/soc/sof/intel/hda.c

```diff
@@ -632,6 +632,23 @@ static int check_nhlt_dmic(struct snd_sof_dev *sdev)
 	return dmic_num;
 }
 
+static int check_nhlt_ssp_mask(struct snd_sof_dev *sdev)
+{
+	struct nhlt_acpi_table *nhlt;
+	int ssp_mask = 0;
+
+	nhlt = intel_nhlt_init(sdev->dev);
+	if (nhlt && intel_nhlt_has_endpoint_type(nhlt, NHLT_LINK_SSP)) {
+		ssp_mask = intel_nhlt_ssp_endpoint_mask(nhlt, NHLT_DEVICE_I2S);
+		if (ssp_mask)
+			dev_info(sdev->dev, "NHLT_DEVICE_I2S detected, ssp_mask %#x\n", ssp_mask);
+
+		intel_nhlt_free(nhlt);
```

Comment:
intel_nhlt_free() won't be called if intel_nhlt_has_endpoint_type(nhlt, NHLT_LINK_SSP) returns 0.
Maybe we can?
```
nhlt = intel_nhlt_init(sdev->dev);
if (!nhlt)
    return ssp_mask;

ssp_mask = intel_nhlt_ssp_endpoint_mask(nhlt, NHLT_DEVICE_I2S);
if (ssp_mask)
    dev_info(sdev->dev, "NHLT_DEVICE_I2S detected, ssp_mask %#x\n", ssp_mask);
intel_nhlt_free(nhlt);

return ssp_mask;
````

Reply:
good catch @libinyang. This needs to be modified indeed. Thanks for the feedback, much appreciated.

Reply:
done